tests/mocha/astnode_test.js

```diff
@@ -333,53 +349,163 @@ suite('ASTNode', function() {
     });
 
     suite('Out', function() {
-      test('topBlock', function() {
-        var node = Blockly.ASTNode.createBlockNode(this.blocks.A);
-        var outNode = node.out();
-        assertEquals(outNode.getLocation(), this.blocks.A);
-      });
-      test('nestedBlock', function() {
-        var node = Blockly.ASTNode.createBlockNode(this.blocks.C);
-        var outNode = node.out();
-        assertEquals(outNode.getLocation(), this.blocks.B.inputList[1].connection);
+      setup(function() {
+        Blockly.defineBlocksWithJsonArray([{
+          "type": "start_block",
+          "message0": "",
+          "nextStatement": null,
+          "colour": 230,
+          "tooltip": "",
+          "helpUrl": ""
+        },
+        {
+          "type": "output_next",
+          "message0": "",
+          "output": null,
+          "colour": 230,
+          "tooltip": "",
+          "helpUrl": "",
+          "nextStatement": null
+        }]);
+        var startBlock = this.workspace.newBlock('start_block');
+        var secondBlock = this.workspace.newBlock('input_statement');
+        var outputNextBlock = this.workspace.newBlock('output_next');
+        var fieldInput = this.workspace.newBlock('field_input');
+
+        startBlock.nextConnection.connect(secondBlock.previousConnection);
+        secondBlock.inputList[0].connection
+            .connect(outputNextBlock.outputConnection);
+        this.blocks.G = startBlock;
```

Comment:
Single-letter names for these blocks are confusing--I can't tell while reading the tests what the blocks refer to.  Pick more descriptive names.

Reply:
done